app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt

```diff
@@ -278,6 +291,9 @@ class BrowserTabViewModel(
     val title: String?
         get() = site?.title
 
+    private lateinit var permissionOrigin: String
+    private lateinit var permissionCallback: GeolocationPermissions.Callback
```

Comment:
This won't work because it's a property and must be initialized. We could initialize them with `emptyString` and a `FakeCallback` or initialize them as null. Making them nullable forces us to always check for nullability, which is the same we have to do now and we wouldn't be adding any improvement. Do you see a better option?

Reply:
>Making them nullable forces us to always check for nullability, which is the same we have to do now 

I think the way we do it now is wrong, in that because it's lateinit, it'll never be null. lint is telling us that.
<img width="837" alt="Screen Shot 2020-08-14 at 5 11 18 PM" src="https://user-images.githubusercontent.com/1336281/90269900-489a2400-de51-11ea-9476-0a1a5426048a.png">

That null check isn't actually keeping us safe. if we access that variable and it wasn't initialized, the app will crash. I think initializing to `nullable` is the least likely to bite us later

Reply:
Any update on this query?